core/src/data/clientGeoJsonSource.cpp

```diff
@@ -77,6 +78,83 @@ ClientGeoJsonSource::ClientGeoJsonSource(std::shared_ptr<Platform> _platform,
 
 ClientGeoJsonSource::~ClientGeoJsonSource() {}
 
+struct add_centroid {
+
+    geometry::point<double>& pt;
```

Comment:
nit: we might be able to find a better name for `pt`, what about `centroidPoint`?